### ISSUE TITLE
(🐞) false unused configs warning when using `no_incremental`

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#16032]
```toml
[tool.mypy]
warn_unused_configs = true

[[tool.mypy.overrides]]
module = ['b.*']
```

```console
👉 mypy a
Success: no issues found in 1 source file
👉 mypy a --no-incremental
Warning: unused section(s) in pyproject.toml: module = ['b.*']
Success: no issues found in 1 source file
```
